### PR TITLE
Accept only ae-encoded addresses, check its length

### PR DESCRIPTION
**Pull request by davidyuk**
I'm proposing to
- ensure that address length is correct to faster find errors
- accept only ae-encoded addresses to simplify interface and validation (still able to create addresses internally as byte arrays (to avoid extra encoding/decoding) by passing `raw` flag)

### CODE REVIEW

File: src/types/FateBytes.js [hunk 1]
```diff
@@ -23,6 +23,10 @@ class FateBytes extends FateData {
 
     // not implemented yet
```
Comment (dincho) [inline src/types/FateBytes.js:24]: could you please also remove this comment as it's now implemented

**Review by dincho — CHANGES_REQUESTED** [review]
I disagree with the proposed implementation as switching logic based on boolean arguments is considered bad practice in general.

Length validation is OK, and I would accept it

File: src/Serializer.js [hunk 2]
```diff
@@ -40,7 +40,7 @@ class Serializer {
         }
 
         // drop FateType prefix
-        const typeName = data.constructor.name.slice(4)
+        const typeName = data.typeName || data.constructor.name.slice(4)
```
Comment (davidyuk) [inline src/Serializer.js:43]: having two classes with the same class name is a tricky thing, proposing this as a workaround

File: src/types/FateAddresses.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+const { FateAddress, FateAddressRaw } = require('./FateAddress.js')
+
+module.exports = [
+    ['Account', 'account_address', 'ak'],
+    ['Channel', 'channel_address', 'ch'],
+    ['Contract', 'contract_address', 'ct'],
+    ['Oracle', 'oracle_address', 'ok'],
+    ['OracleQuery', 'oracle_query_address', 'oq'],
+].reduce((prev, [name, ...args]) => {
```
Comment (davidyuk) [inline src/types/FateAddresses.js:9]: Combining these types into a single file to simplify the making of changes, not sure is it acceptable. But I don't think these types got a custom behavior in a foreseen future.

**Review by dincho — CHANGES_REQUESTED** [review]
too much meta

**Comment by dincho** [comment]
@davidyuk would you benefit from changes in #89 as you mentioned there ?

**Comment by dincho** [comment]
sorry for the jaded review, thanks for your time

**Review by dincho — APPROVED** [review]

**Comment by davidyuk** [comment]
it is fine, was interesting to know your opinion about the above stuff

**Pull request merged**